src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -20,18 +21,24 @@ public abstract class Renderer : IDisposable
         private readonly Dictionary<int, ComponentState> _componentStateById = new Dictionary<int, ComponentState>();
         private readonly RenderBatchBuilder _batchBuilder = new RenderBatchBuilder();
         private readonly Dictionary<int, EventHandlerInvoker> _eventBindings = new Dictionary<int, EventHandlerInvoker>();
+        private RendererSynchronizationContext _synchronizationContext = new RendererSynchronizationContext();
 
         private int _nextComponentId = 0; // TODO: change to 'long' when Mono .NET->JS interop supports it
         private bool _isBatchInProgress;
         private int _lastEventHandlerId = 0;
         private List<Task> _pendingTasks;
 
-        // We need to introduce locking as we don't know if we are executing
-        // under a synchronization context that limits the ammount of concurrency
-        // that can happen when async callbacks are executed.
-        // As a result, we have to protect the _pendingTask list and the
-        // _batchBuilder render queue from concurrent modifications.
-        private object _asyncWorkLock = new object();
+        public event UnhandledExceptionEventHandler UnhandledSynchronizationException
```

Comment:
This is just temporary until we have @pranavkm improved error handling

Reply:
Sounds good.